src/Files.App/Helpers/DragZoneHelper.cs

```diff
@@ -0,0 +1,166 @@
+					var y = draggingZonesY[i].ToArray();
+					var xSubtrahend = new Range(nonDraggingZone.X, nonDraggingZone.X + nonDraggingZone.Width);
+					var ySubtrahend = new Range(nonDraggingZone.Y, nonDraggingZone.Y + nonDraggingZone.Height);
+					var xResult = (x - xSubtrahend).ToArray();
+					if (xResult.Length is 1 && xResult[0] == x)
+						continue;
+					var yResult = (y - ySubtrahend).ToArray();
+					switch (xResult.Length)
+					{
+						case 0:
+							draggingZonesY[i] = yResult;
+							break;
+						case 1:
+							draggingZonesX.RemoveAt(i);
+							draggingZonesY.RemoveAt(i);
+							if (xResult[0].Lower == x.Lower)
+							{
+								draggingZonesY.InsertRange(i, new[] { y, yResult });
+								draggingZonesX.InsertRange(i, new[]
+								{
+									x with { Upper = xResult[0].Upper },
+									x with { Lower = xSubtrahend.Lower }
+								});
+							}
+							else // xResult[0].Upper == x.Upper
+							{
+								draggingZonesY.InsertRange(i, new[] { yResult, y });
+								draggingZonesX.InsertRange(i, new[]
+								{
+									x with { Upper = xSubtrahend.Upper },
+									x with { Lower = xResult[0].Lower }
+								});
+							}
+							++i;
+							break;
+						case 2:
+							draggingZonesX.RemoveAt(i);
+							draggingZonesY.RemoveAt(i);
+							draggingZonesY.InsertRange(i, new[] { y, yResult, y });
+							draggingZonesX.InsertRange(i, new[]
+							{
+								x with { Upper = xResult[0].Upper },
+								xSubtrahend,
+								x with { Lower = xResult[1].Lower }
+							});
+							++i;
+							++i;
+							break;
+					}
+				}
+			}
+
+			var rects = draggingZonesX
+				.SelectMany((rangeX, i) => draggingZonesY[i]
+					.Select(rangeY => new RectInt32(rangeX.Lower, rangeY.Lower, rangeX.Distance, rangeY.Distance)))
+				.OrderBy(t => t.Y)
+				.ThenBy(t => t.X).ToList();
+			for (var i = 0; i < rects.Count - 1; ++i)
+			{
+				var now = rects[i];
+				var next = rects[i + 1];
+				if (now.Height == next.Height && now.X + now.Width == next.X)
+				{
+					rects.RemoveRange(i, 2);
+					rects.Insert(i, now with { Width = now.Width + next.Width });
+				}
+			}
+
+			return rects;
+		}
+
+		/// <summary>
+		/// Set dragging-zones of title bar
+		/// </summary>
+		/// <param name="window"></param>
+		/// <param name="dragZoneHeight"></param>
+		/// <param name="dragZoneLeftIndent"></param>
+		/// <param name="nonDraggingZones"></param>
+		public static void SetDragZones(Window window, int dragZoneHeight = 40, int dragZoneLeftIndent = 0, IEnumerable<RectInt32>? nonDraggingZones = null)
+		{
+			var hWnd = WindowNative.GetWindowHandle(window);
+			var windowId = Win32Interop.GetWindowIdFromWindow(hWnd);
+			var appWindow = AppWindow.GetFromWindowId(windowId);
+			var scaleAdjustment = GetScaleAdjustment(window);
+			var windowWidth = (int)(appWindow.Size.Width / scaleAdjustment);
+			nonDraggingZones ??= Array.Empty<RectInt32>();
+#if DEBUG   // subtract the toolbar area (center-top in window), only in DEBUG mode.
```

Comment:
Might be a good idea to move the comment to a new line.

Reply:
Done